tests/src/test/scala/com/nvidia/spark/rapids/SparkQueryCompareTestSuite.scala

```diff
@@ -249,6 +249,57 @@ trait SparkQueryCompareTestSuite extends FunSuite with Arm {
     }
   }
 
+  def writeOnCpuAndGpuWithCapture(df: SparkSession => DataFrame,
```

Comment:
its a bit odd this is called writeOn when this function doesn't do the write, it depends on the function passed in.  Perhaps renamed to be something more generic executeDF or something like that.